posydon/interpolation/profile_interpolation.py

```diff
@@ -0,0 +1,553 @@
+"""Module for performing final profile interpolation
+"""
+
+__authors__ = [
+    "Elizabeth Teng <elizabethteng@u.northwestern.edu>"
+]
+
+import pickle
+import matplotlib.pyplot as plt
+
+# POSYDON
+from posydon.grids.psygrid import PSyGrid
+from posydon.interpolation.IF_interpolation import IFInterpolator
+
+# Math and ML
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+tf.get_logger().setLevel('ERROR')
+from tensorflow.keras import layers, losses, models, optimizers
+from sklearn.decomposition import PCA
+from scipy.interpolate import interp1d
+
+
+class CompileData:
+
+    def __init__(self, train_path, valid_path,
+                 profile_names=['radius','logRho', 'x_mass_fraction_H','y_mass_fraction_He',
+                                 'z_mass_fraction_metals','omega','energy']):
+        """Extracts profile data from '.h5' grid files and saves to file.
+        Args:
+            train_path (str) : path/name of '.h5' file for training data.
+            valid_path (str) : path/name of '.h5' file for testing/validation data.
+            profile_names (array-like) : list of profile quantities to be extracted.
+        """
+        self.names = profile_names
+        
+        try:
+            print(valid.final_values["S1_state"][0])
```

Comment:
Did I missed it, or is the variable `valid` not defined before that try? It will get set in line 45.

Reply:
yes, you're right! thanks for catching that